nym-api/src/unstable_routes/account.rs

```diff
@@ -0,0 +1,199 @@
+// Copyright 2025 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: GPL-3.0-only
+
+use crate::{
+    node_status_api::models::{AxumErrorResponse, AxumResult},
+    support::http::state::AppState,
+    unstable_routes::models::{
+        NymVestingAccount, NyxAccountDelegationDetails, NyxAccountDelegationRewardDetails,
+        NyxAccountDetails,
+    },
+};
+use axum::{
+    extract::{Path, State},
+    routing::get,
+    Json, Router,
+};
+use cosmwasm_std::{Coin, Decimal};
+use nym_validator_client::nyxd::AccountId;
+use serde::{Deserialize, Serialize};
+use std::{
+    collections::{HashMap, HashSet},
+    str::FromStr,
+};
+use tracing::{error, instrument, warn};
+use utoipa::ToSchema;
+
+pub(crate) fn routes() -> Router<AppState> {
+    Router::new().route("/:address", get(address))
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, ToSchema, utoipa::IntoParams)]
+pub struct AddressQueryParam {
+    #[serde(default)]
+    pub address: String,
+}
+
+#[utoipa::path(
+    tag = "Unstable",
+    get,
+    path = "/{address}",
+    context_path = "/v1/unstable/account",
+    responses(
+        (status = 200, body = NyxAccountDetails)
+    ),
+    params(AddressQueryParam)
+)]
+#[instrument(level = "info", skip_all, fields(address=address))]
+async fn address(
+    Path(AddressQueryParam { address }): Path<AddressQueryParam>,
+    State(state): State<AppState>,
+) -> AxumResult<Json<NyxAccountDetails>> {
+    let account_id = AccountId::from_str(&address).map_err(|err| {
+        error!("{err}");
+        AxumErrorResponse::not_found(&address)
+    })?;
+    let base_denom = &state.network_details().network.chain_details.mix_denom.base;
+
+    let state_client = &state.nyxd_client;
+    let mut total_value = 0u128;
+
+    // 1. get balances of chain tokens
+    let balance = state_client
+        .get_address_balance(&account_id, base_denom)
+        .await?
+        .unwrap_or_else(|| nym_validator_client::nyxd::Coin::new(0u128, base_denom));
+    total_value += balance.amount;
+
+    // 2. get list of delegations (history)
+    let og_delegations = state_client
+        .get_all_delegator_delegations(&account_id)
+        .await?;
+
+    let delegated_to_nodes = og_delegations
+        .iter()
+        .map(|d| d.node_id)
+        .collect::<HashSet<_>>();
+
+    let mut operator_rewards = 0;
+    let nym_nodes = state
+        .nym_contract_cache()
+        .all_cached_nym_nodes()
+        .await
+        .ok_or_else(AxumErrorResponse::service_unavailable)?
+        .iter()
+        .filter_map(|node_details| {
```

Comment:
I thought I already left a comment about it - there's no point in iterating the whole cache, just do a `.find()` because there can be at most one node that belongs to particular address